plugins/repository-gcs/src/main/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageService.java

```diff
@@ -19,74 +19,92 @@
 
 package org.elasticsearch.repositories.gcs;
 
-import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
 import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
-import com.google.api.client.http.HttpBackOffIOExceptionHandler;
-import com.google.api.client.http.HttpBackOffUnsuccessfulResponseHandler;
-import com.google.api.client.http.HttpRequest;
-import com.google.api.client.http.HttpRequestInitializer;
-import com.google.api.client.http.HttpTransport;
-import com.google.api.client.http.HttpUnsuccessfulResponseHandler;
-import com.google.api.client.json.jackson2.JacksonFactory;
-import com.google.api.client.util.ExponentialBackOff;
-import com.google.api.services.storage.Storage;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.gax.retrying.RetrySettings;
+import com.google.cloud.http.HttpTransportOptions;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+
 import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.collect.MapBuilder;
 import org.elasticsearch.common.component.AbstractComponent;
 import org.elasticsearch.common.unit.TimeValue;
 import org.elasticsearch.env.Environment;
+import org.threeten.bp.Duration;
 
 import java.io.IOException;
+import java.security.GeneralSecurityException;
 import java.util.Map;
 
 public class GoogleCloudStorageService extends AbstractComponent {
 
     /** Clients settings identified by client name. */
     private final Map<String, GoogleCloudStorageClientSettings> clientsSettings;
+    private final RetrySettings retrySettings;
 
-    public GoogleCloudStorageService(final Environment environment, final Map<String, GoogleCloudStorageClientSettings> clientsSettings) {
+    public GoogleCloudStorageService(Environment environment, Map<String, GoogleCloudStorageClientSettings> clientsSettings) {
         super(environment.settings());
         this.clientsSettings = clientsSettings;
+        this.retrySettings = RetrySettings.newBuilder()
+                .setInitialRetryDelay(Duration.ofMillis(100))
+                .setMaxRetryDelay(Duration.ofMillis(6000))
+                .setTotalTimeout(Duration.ofMillis(900000))
+                .setRetryDelayMultiplier(1.5d)
+                .setJittered(true)
+                .build();
     }
 
     /**
      * Creates a client that can be used to manage Google Cloud Storage objects.
      *
-     * @param clientName name of client settings to use from secure settings
+     * @param clientName
+     *            name of client settings to use from secure settings
      * @return a Client instance that can be used to manage Storage objects
      */
-    public Storage createClient(final String clientName) throws Exception {
+    public Storage createClient(String clientName) throws GeneralSecurityException, IOException {
```

Comment:
Yes, sorry for this. The line changed because the exception list changed, and the IDE automatically removed the final from parameters (as this is not a rule in my IDE config).